Open Judge System/Web/OJS.Web/Controllers/TempController.cs

```diff
@@ -139,5 +141,79 @@ public ActionResult RetestCompileTimeoutSubmissionsFromCSharpDbAdvancedExam10Dec
 
             return this.Content($"Successfully enqueued {submissionIds.Count()} submissions for retesting.");
         }
+
+        public ActionResult MigrateProblemsToIndividualGroups()
+        {
+            var contests = this.Data.Contests
+                .AllWithDeleted()
+                .Include(c => c.Problems)
+                .Where(c => c.NumberOfProblemGroups <= 1 && !c.ProblemGroups.Any())
+                .ToList();
+
+            var output = new StringBuilder();
+            output.Append($"Done! contests affected: {contests.Count} <ol>");
+
+            foreach (var contest in contests)
+            {
+                foreach (var problem in contest.Problems)
+                {
+                    var problemGroup = new ProblemGroup
```

Comment:
What's the point of creating problem group for each problem?

Reply:
To be semantically right as we use Problem Group for grouping problems that can be interchangeable in the Online contest and we want to move towards this logic in every contest. But in the current contests every problem is created as unique and cannot be changed by another, that's why we want them to be in a separate groups.